### PR TITLE
ci: introduce Release Please and PR linting workflows

### LINKED ISSUE #6: Automate release process

### PR DESCRIPTION
**Pull request by thisthat**
## This PR

- adds workflow for linting PR title
- adds workflow to automate the release process with Release Please

### Related Issues

- fixes #6 

### Notes

Builds on top of https://github.com/open-feature/python-sdk-contrib/pull/36

### How to test

Manually tested on my fork without the publishing step.

Hooks:
- Release Please PR: https://github.com/thisthat/python-sdk-contrib/pull/14
- Release action: https://github.com/thisthat/python-sdk-contrib/actions/runs/7995359933/job/21835517843#step:10:5

Provider:
- Release Please PR: https://github.com/thisthat/python-sdk-contrib/pull/16
- Release action: https://github.com/thisthat/python-sdk-contrib/actions/runs/7995418337/job/21835711626#step:4:10

Full list of releases: https://github.com/thisthat/python-sdk-contrib/releases

### CODE REVIEW

File: release-please-config.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      "type": "style",
+      "hidden": true,
+      "section": "🎨 Styling"
+    },
+    {
+      "type": "test",
+      "hidden": true,
+      "section": "🧪 Tests"
+    }
+  ],
```
Comment (federicobond) [inline release-please-config.json:82]: TODO: we should add these sections to the python-sdk repo config, to standarize both changelogs.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,27 +28,20 @@ We use `pytest` for our unit testing, making use of `parametrized` to inject cas
...
-
 ## Pull Request
 
 All contributions to the OpenFeature project are welcome via GitHub pull requests.
 
 To create a new PR, you will need to first fork the GitHub repository and clone upstream.
 
 ```bash
-git clone https://github.com/open-feature/python-sdk-contrib.git openfeature-python-sdk-contrib
+git clone https://github.com/open-feature/python-sdk-contrib.git
```
Comment (federicobond) [inline CONTRIBUTING.md:38]: I think the original rename here is a nice touch, we do it in the CONTRIBUTING.md file of the python-sdk and other OpenFeature repos too. Can we keep it?

File: .github/workflows/lint-pr.yml [hunk 3]
```diff
@@ -13,13 +13,38 @@ on:
       - synchronize
 
 permissions:
-  pull-requests: read
+  pull-requests: write
```
Comment (federicobond) [inline .github/workflows/lint-pr.yml:16]: The JS SDK apparently manages to get by without this permission. Can we confirm it's required?

See https://github.com/open-feature/js-sdk/blob/main/.github/workflows/lint-pr.yml

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        with:
+          command: manifest
+          token: ${{secrets.GITHUB_TOKEN}}
+          default-branch: main
+          signoff: "OpenFeature Bot <109696520+openfeaturebot@users.noreply.github.com>"
+      - name: Dump Release Please Output
+        env:
+          RELEASE_PLEASE_OUTPUT: ${{ toJson(steps.release.outputs) }}
+        run: |
+          echo "$RELEASE_PLEASE_OUTPUT"
```
Comment (federicobond) [inline .github/workflows/release.yml:28]: Should we keep this dump step or is it only for debugging the first releases?

**Review by federicobond — COMMENTED** [review]
Awesome work @thisthat! I added just a couple of minor comments.

**Comment by federicobond** [comment]
@thisthat fix for failing build is [here](https://github.com/open-feature/python-sdk-contrib/pull/39).

File: .github/workflows/release.yml (hunk 4 above)
Comment (thisthat) [reply .github/workflows/release.yml:28]: I find this very helpful to debug issues when something goes off at release time 😅 
we can remove it if you prefer tho

File: .github/workflows/lint-pr.yml (hunk 3 above)
Comment (thisthat) [reply .github/workflows/lint-pr.yml:16]: 👀 I will look into it

File: release-please-config.json (hunk 1 above)
Comment (thisthat) [reply release-please-config.json:82]: I'd say yes, it makes sense. They are also used in the go, js, and java sdks

File: .github/workflows/lint-pr.yml (hunk 3 above)
Comment (thisthat) [reply .github/workflows/lint-pr.yml:16]: from the docs (https://github.com/marocchino/sticky-pull-request-comment#basic) it is needed.
Looking into GH Action docs (see https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-pull-requests), adding a comment requires this permission.
I'll create an issue for the JS-SDK :)

Comment (thisthat) [reply .github/workflows/lint-pr.yml:16]: created https://github.com/open-feature/js-sdk/issues/827

File: CONTRIBUTING.md (hunk 2 above)
Comment (thisthat) [reply CONTRIBUTING.md]: reverted in [bc93bcc](https://github.com/open-feature/python-sdk-contrib/pull/38/commits/bc93bcc79ab1cc676a2c1265dd5f0e715eaa7d64)

File: .github/workflows/release.yml (hunk 4 above)
Comment (federicobond) [reply .github/workflows/release.yml:28]: No, it's fine! Just wanted to have a record of why it's there.

**Review by federicobond — APPROVED** [review]

**Pull request merged**